pkg/alertmanager/operator.go

```diff
@@ -760,6 +760,7 @@ func createSSetInputHash(a monitoringv1.Alertmanager, c Config, tlsAssets *opera
 		AlertmanagerAnnotations map[string]string
 		AlertmanagerGeneration  int64
 		AlertmanagerWebHTTP2    *bool
+		AlertmanagerClusterTLS  *monitoringv1.ClusterTLSConfig
```

Comment:
The `AlertmanagerGeneration` value should already change when the user updates the cluster TLS config. The things which aren't triggering a rollout of the statefulset are:
* Updates to the data in the secrets/configmaps holding the TLS materials.
* Updates to the TLS files when using `caFile`, `certFile` or `keyFile`.

The latter is harder (impossible?) for the operator to detect so tackling only the first case is fine.